### DISCUSSION TITLE
How to set the Blob type to audio/wav

### DISCUSSION THREAD

**Discussion by possibly** [samhirtarif/react-audio-recorder#75] [Q&A]
When using MediaRecorder, the type of the Blob file can be set to formats other than webm. Is there a way to set the Blob type to audio/wav?

I have the AudioRecorder component:
```javascript
<AudioRecorder
            onRecordingComplete={addAudioElement}
            downloadFileExtension="wav"
            ...
```

addAudioElement looks like:
```javascript
const addAudioElement = async (blob: Blob) => {
    const audioArrayBuffer = await blob.arrayBuffer();
    const base64AudioData = btoa(String.fromCharCode(...new Uint8Array(audioArrayBuffer)))
    console.log(base64AudioData)
  };
``` 

Using a tool like https://base64.guru/converter/decode/audio to decode the base64 encoded audio back into audio shows that the audio is in webm format.

Is it possible to simply encode the data chunks from the media recorder as audio/wav without worry about cross origin isolation since I am not interested in downloading the audio?

**Comment by samhirtarif** [comment]
@possibly - This package uses the package [@ffmpeg/ffmpeg](https://www.npmjs.com/package/%40ffmpeg%2Fffmpeg) to convert audio from `webm` to `wav` or `mp3`
But since [@ffmpeg/ffmpeg](https://www.npmjs.com/package/%40ffmpeg%2Fffmpeg) uses `SharedArrayBuffer` that requires a site to be [cross-origin isolated](https://web.dev/cross-origin-isolation-guide/), this conversion only works when your site is cross-origin isolated, otherwise the downloaded format is always `webm`

<img width="729" alt="image" src="https://github.com/samhirtarif/react-audio-recorder/assets/89391183/9ac6349a-1a13-4acc-a9eb-57568373f603">
<br /><br />
In case you are using vite, I can share a code snippet with you that would at least make your development server cross-origion isolated

**Reply by possibly** [reply]
Hey @samhirtarif ! Thanks for getting back to me.

I saw that, and that limitation. As an alternative, I've seen snippets like the following floating around:![image](https://github.com/samhirtarif/react-audio-recorder/assets/1417341/1993ecfa-5a0d-4a40-96bd-60adc1d3baad)

From my limited understanding, it seems like there is some way to take the audio chunks in some raw format and, right before putting them into the Blob, encode them in the desired format there?

Is this technique applicable to this project, and for the wav file format?

**Reply by samhirtarif** [reply]
@possibly - This is the approach that was initially used in this package to do the file type conversions as well, but to no avail.

The downloaded files were still in the `webm` format, despite creating a new `Blob` with appropriate `mimetype `and appending the proper extension to the downloaded file.

**Comment by possibly** [comment]
@samhirtarif Ah I see.

I also tried passing my desired `mimetype` in the `mediaRecorderOptions` parameter, similar to whats in [the demo](https://stackblitz.com/edit/react-ts-cc5l47?file=App.tsx), but both Chrome and Firefox threw errors.

Thanks for looking into the available options and providing the ffmpeg one!
